lib/instrumentation/langchain/vectorstore.js

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2024 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+const {
+  AI: { LANGCHAIN }
+} = require('../../metrics/names')
+const { LangChainVectorSearch, LangChainVectorSearchResult } = require('../../llm-events/langchain')
+const { recordEvent, shouldSkipInstrumentation } = require('./common')
+const { DESTINATIONS } = require('../../config/attribute-filter')
+const { RecorderSpec } = require('../../shim/specs')
+const LlmErrorMessage = require('../../llm-events/error-message')
+
+/**
+ * Generates a LangChainVectorSearch for entire search request.
+ * Also iterates over documents in output and generates a
+ * LangChainVectorSearchResult for each document.
+ *
+ * @param {string} request vector search query
+ * @param {number} k vector search top k
+ * @param {object} output vector search documents
+ * @param {Agent} agent NR agent instance
+ * @param {TraceSegment} segment active segment from vector search
+ * @param {string} pkgVersion langchain version
+ * @param {err} err if it exists
+ */
+// eslint-disable-next-line max-params
+function recordVectorSearch(request, k, output, agent, segment, pkgVersion, err) {
+  const vectorSearch = new LangChainVectorSearch({
+    agent,
+    segment,
+    query: request,
+    k: k,
+    documents: output,
+    error: err !== null
+  })
+
+  output.forEach((document, sequence) => {
+    const vectorSearchResult = new LangChainVectorSearchResult({
+      agent,
+      segment,
+      metadata: document.metadata,
+      pageContent: document.pageContent,
+      sequence,
+      id: vectorSearch.id
+    })
+
+    recordEvent({
+      agent,
+      type: 'LlmVectorSearchResult',
+      pkgVersion,
+      msg: vectorSearchResult
+    })
+  })
+
+  recordEvent({ agent, type: 'LlmVectorSearch', pkgVersion, msg: vectorSearch })
```

Comment:
does this have to be recorded after the LlmVectorSearchResult? not a big deal it just threw me off at first why the event was made but logging of event was delayed